### PR TITLE
PathFollow2D unit test 

### PR DESCRIPTION
**Pull request by angad-k**
part of #43440
Adds a unit test for PathFollow2D node.
Tests the following : 

- [x] Sampling with unit offset
- [x] Sampling with offset
- [x] Behavior on removal of point from curve
- [x] Behavior on setting up an h_offset and a v_offset
- [x] Behavior when unit offset is below `0` and above `1`
- [x] Behavior when offset is below `0` and above the max value

Haven't run the tests yet. Will squash the commits when I run the tests and all the checks pass.

### CODE REVIEW

File: tests/test_path_follow_2d.h [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+#include "scene/2d/path_2d.h"
+#include "scene/resources/curve.h"
+
+#include "tests/test_macros.h"
+
+namespace TestPathFollow2D {
+
+TEST_CASE("[PathFollow2D] Sampling with unit offset") {
+	const Ref<Curve2D> curve = memnew(Curve2D());
```
Comment (Xrayez) [inline tests/test_path_follow_2d.h:42]:
```suggestion
	const Ref<Curve2D> &curve = memnew(Curve2D());
```

See also https://github.com/godotengine/godot/pull/43731#discussion_r578037411.

The same applies to other `const Ref<T>` occurrences.

Comment (angad-k) [reply tests/test_path_follow_2d.h]: @Xrayez addressed the review

Comment (angad-k) [reply tests/test_path_follow_2d.h]: If everything's good now, should I squash the commits? Also, should I add any other test cases?

**Review by Xrayez — APPROVED** [review]
I haven't used those nodes extensively, but tests looks good to me!

Need to squash commits.

**Comment by angad-k** [comment]
Squashed the commits

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**